jax/lax/lax_control_flow.py

```diff
@@ -947,6 +948,10 @@ def _scan_partial_eval(trace, *tracers, forward, length, num_consts, num_carry,
   const_avals_1 = [raise_to_shaped(core.get_aval(c)) for c in consts_1]
   in_avals_1 = [core.abstract_unit] * num_consts + jaxpr_1.in_avals[num_consts:]
   out_avals_1 = [core.abstract_unit if pv is None else pv for pv, c in out_pvals_1]
+
+  for var in untyped_jaxpr_1.invars[:num_consts]:
```

Comment:
This mutation is a strange thing to have to do, but I think I understand it. Without it probably the type checking assertions would fail.

Reply:
Yes, I have no justification for this, other than that it makes the assertion pass! I really think we should add a comment here explaining the need for this. If somebody that actually understands the code could provide me with an explanation, that would be great!